src/py/requirements.txt

```diff
@@ -1,8 +1,12 @@
 # You should not edit this file directly.  Instead, you should edit one of the following files (src/py/requirements.txt.in) and run make src/py/requirements.txt
-click==7.1.2
+Authlib==0.15.3
 Flask==1.1.2
-itsdangerous==1.1.0
 Jinja2==2.11.2
 MarkupSafe==1.1.1
 SQLAlchemy==1.3.22
 Werkzeug==1.0.1
+click==7.1.2
+itsdangerous==1.1.0
+python-dotenv==0.15.0
+requests==2.24.0
+six==1.15.0
```

Comment:
You should edit `src/py/requirements.txt.in`, add in python-dotenv and Authlib, and run `make -j pin-requirements`

Reply:
oh ok so in the future we add new requirements to requirements.txt.ini and make -j pin-requirements will add them to requirements.txt? Can I ask what the benefit is of structuring it this way?

Reply:
Yeah, we are tracking the requirements we actually need.  The transitive dependencies are pinned but kept separate.